### PR TITLE
Update login

### PR DESCRIPTION
**Pull request by Rittamx031**
## Related Tickets
[- ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/72833)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

### CODE REVIEW

**Comment by Rittamx031** [comment]
ready

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -13,24 +13,38 @@ def default_url_options
...
 
-  def check_user_role
-    return true unless admin?
-
-    flash[:warning] = t("controllers.errors.only_user_role")
+  def check_user_role?
+    if current_user.admin?
+      flash[:danger] = t("controllers.errors.only_user_role")
+    end
+    !current_user.admin?
```
Comment (ghost) [inline app/controllers/application_controller.rb]:
```suggestion
    current_user.admin?
```

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -13,24 +13,38 @@ def default_url_options
   end
 
   def check_user
-    redirect_to login_url unless check_login && check_user_role
+    if !check_login?
+      redirect_to login_path
+    elsif !check_user_role?
```
Comment (ghost) [inline app/controllers/application_controller.rb]:
```suggestion
    elsif check_user_role?
```

Comment (ghost) [reply app/controllers/application_controller.rb]: Như này đc ko nhỉ

File: app/controllers/relationships_controller.rb [hunk 3]
```diff
@@ -1,15 +1,17 @@
 class RelationshipsController < ApplicationController
-  before_action :logged_in_user
+  before_action :authenticate_user!
   before_action :load_tour, only: %i(create destroy)
 
   def create
     current_user.following_tour(@tour)
```
Comment (ghost) [inline app/controllers/relationships_controller.rb]: Ko bắt trường hợp failed à em

File: app/controllers/relationships_controller.rb [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
     flash.now[:success] = t("tours.follow_success")
+    redirect_to tours_path
   end
 
   def destroy
     current_user.unfollow_tour(@tour)
```
Comment (ghost) [inline app/controllers/relationships_controller.rb]: Đây nx

File: app/models/user.rb [hunk 5]
```diff
@@ -1,8 +1,11 @@
 class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :rememberable, :validatable
+
   SIGNUP_PARAMS = [:username, :phone, :email,
                   :password, :avatar,
                   :password_confirmation].freeze
-  attr_accessor :remember_token
+  # attr_accessor :remember_token
```
Comment (ghost) [inline app/models/user.rb]: Ko dùng nx thì bỏ luôn em nhé

File: app/views/admin/layouts/_header.html.erb [hunk 6]
```diff
@@ -21,35 +21,22 @@
...
+          <%= image_tag current_user.avatar.variant(:avatar_icon), alt: "Profile", class: "rounded-circle" if current_user.avatar&.attached? %>
+          <span class="d-none d-md-block dropdown-toggle ps-2">
+            <%= current_user.username %>
+          </span>
+        </a>
+        <ul class="dropdown-menu dropdown-menu-end dropdown-menu-arrow profile">
+          <li>
+            <%= link_to signout_path, data: { turbo_method: :delete }, class: "dropdown-item d-flex align-items-center" do %>
+              <i class="bi bi-box-arrow-right"></i>
+              <span>Đăng xuất</span>
```
Comment (ghost) [inline app/views/admin/layouts/_header.html.erb]: I18n nè

**Review by ghost — COMMENTED** [review]
cmt

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (Rittamx031) [reply app/controllers/application_controller.rb]: dạ khum a

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (Rittamx031) [reply app/controllers/application_controller.rb]: này trả về true nếu ko phải admin a

**Comment by Rittamx031** [comment]
ready

**Comment by ghost** [comment]
Failed CI r

**Comment by Rittamx031** [comment]
ready

**Comment by Rittamx031** [comment]
ready

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ghost** [comment]
merged!

**Pull request merged**